libraries/state_history/abi.cpp

```diff
@@ -418,18 +418,48 @@ extern const char* const state_history_plugin_abi = R"({
                 { "type": "uint16", "name": "max_authority_depth" }
             ]
         },
+        {
+            "name": "chain_config_v1", "fields": [
+                { "type": "uint64", "name": "max_block_net_usage" },
+                { "type": "uint32", "name": "target_block_net_usage_pct" },
+                { "type": "uint32", "name": "max_transaction_net_usage" },
+                { "type": "uint32", "name": "base_per_transaction_net_usage" },
+                { "type": "uint32", "name": "net_usage_leeway" },
+                { "type": "uint32", "name": "context_free_discount_net_usage_num" },
+                { "type": "uint32", "name": "context_free_discount_net_usage_den" },
+                { "type": "uint32", "name": "max_block_cpu_usage" },
+                { "type": "uint32", "name": "target_block_cpu_usage_pct" },
+                { "type": "uint32", "name": "max_transaction_cpu_usage" },
+                { "type": "uint32", "name": "min_transaction_cpu_usage" },
+                { "type": "uint32", "name": "max_transaction_lifetime" },
+                { "type": "uint32", "name": "deferred_trx_expiration_window" },
+                { "type": "uint32", "name": "max_transaction_delay" },
+                { "type": "uint32", "name": "max_inline_action_size" },
+                { "type": "uint16", "name": "max_inline_action_depth" },
+                { "type": "uint16", "name": "max_authority_depth" },
+                { "type": "uint32", "name": "max_action_return_value_size" }
+            ]
+        },
         {
             "name": "global_property_v0", "fields": [
                 { "type": "uint32?", "name": "proposed_schedule_block_num" },
                 { "type": "producer_schedule", "name": "proposed_schedule" },
-                { "type": "chain_config", "name": "configuration" }
+                { "type": "chain_config_v0", "name": "configuration" }
             ]
         },
         {
             "name": "global_property_v1", "fields": [
                 { "type": "uint32?", "name": "proposed_schedule_block_num" },
                 { "type": "producer_authority_schedule", "name": "proposed_schedule" },
-                { "type": "chain_config", "name": "configuration" },
+                { "type": "chain_config_v0", "name": "configuration" },
```

Comment:
Is it necessary to add a new version of `global_property`?  The only thing that changed was `chain_config` which is already a variant.

Reply:
in eos it is structure and its size is changed as per this PR. not sure how state history stuff works though

Reply:
If `state_history` represents a type as a variant, then changes to the type's layout don't need to be propagated up to containing types.

Reply:
@swatanabe-b1 I made changes accordingly. also I believe I need to make similar changes to:
[libraries/abieos/include/eosio/ship_protocol.hpp](https://github.com/EOSIO/abieos/blob/develop/include/eosio/ship_protocol.hpp)